addons/account/models/account_payment.py

```diff
@@ -447,11 +447,13 @@ def _compute_partner_id(self):
     def _compute_outstanding_account_id(self):
         for pay in self:
             if pay.payment_type == 'inbound':
-                pay.outstanding_account_id = (pay.payment_method_line_id.payment_account_id
-                                              or pay.journal_id.company_id.account_journal_payment_debit_account_id)
+                pay.outstanding_account_id = (pay.journal_id.inbound_payment_method_line_ids.payment_account_id
```

Comment:
I think this is not correct, if you have two or more lines with accounts on your journal it would try to put all of them in the Many2one.

Also, I think changing the compute is not needed. We investigated the issue this morning and it seems that the compute `_compute_payment_method_line_id` is not called during the copy in `_create_paired_internal_transfer_payment` because the fields it depends on are already set before the copy.
What is wrong is that the `payment_method_line_id` is thus set to the one from the original journal and not recomputed leading to this incorrect account being set.

I think the best solution here would simply be to mark the field `payment_method_line_id` as copy=False which would lead it to be recomputed when copying a payment. And to recompute the account correctly at the same time.

Reply:
Indeed, you're right, thanks